src/rl78.c

```diff
@@ -40,48 +48,105 @@ static void rl78_set_reset(port_handle_t fd, int mode, int value)
     {
         serial_set_dtr(fd, level);
     }
+    }
 }
 
 int rl78_reset_init(port_handle_t fd, int wait, int baud, int mode, float voltage)
+// Routine for resetting RL78 and initializing it in programming mode
 {
     unsigned char r;
+    //(mode = desired communication mode from terminal input - 1)
+
+   // Determine UART mode from mode argument
     if (MODE_UART_1 == (mode & MODE_UART))
     {
         r = SET_MODE_1WIRE_UART;
         communication_mode = 1;
     }
+    
     else
     {
         r = SET_MODE_2WIRE_UART;
         communication_mode = 2;
     }
     if (4 <= verbose_level)
     {
+        if (mode == 4) // Special case for mode 5
+        {
+            communication_mode = 5;
+            printf("Using communication mode 5 (HW RESET with CTS reading RESET)\n");
+        }
```

Comment:
Don't change `communication_mode` here.  I think we want this to stay either 1 or 2.  You can still output the verbose message, but trigger that based on `(mode & MODE_RESET_INPUT == MODE_RESET_INPUT_TRUE` instead of `mode == 4`.